internal/http/services/owncloud/ocs/handlers/cloud/users/users.go

```diff
@@ -100,3 +103,59 @@ type Users struct {
 type Groups struct {
 	Groups []string `json:"groups" xml:"groups>element"`
 }
+
+func (h *Handler) handleUsers(w http.ResponseWriter, r *http.Request, u *userpb.User) {
+	ctx := r.Context()
+	sublog := appctx.GetLogger(r.Context())
+
+	gc, err := pool.GetGatewayServiceClient(h.gatewayAddr)
+	if err != nil {
+		sublog.Error().Err(err).Msg("error getting gateway client")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	getHomeRes, err := gc.GetHome(ctx, &provider.GetHomeRequest{})
+	if err != nil {
+		sublog.Error().Err(err).Msg("error calling GetHome")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if getHomeRes.Status.Code != rpc.Code_CODE_OK {
+		ocdav.HandleErrorStatus(sublog, w, getHomeRes.Status)
+		return
+	}
+
+	getQuotaRes, err := gc.GetQuota(ctx, &gateway.GetQuotaRequest{
+		Ref: &provider.Reference{
```

Comment:
Since we discard Ref in `gateway.storageprovider.GetQuota` anyway, maybe we can save the extra call going to GetHome? And pass Ref as empty.

Reply:
No, we need to look up the users home storage space before making the GetQuota call at the gateway. It will forward the request to the storage provider determined by the storageid in the reference returned by GetHome().

the comment in the gateway
```
		//Ref:    req.GetRef(), // TODO send which storage space ... or root
```
Will make more sense when the storage registry and storage providers become more dynamic, eg. when a storage provider can really manage multiple storage spaces. Then the storage provider needs to know which storage space is supposed to be queried.